replicators/src/mysql_connector/connector.rs

```diff
@@ -128,8 +132,28 @@ impl MySqlBinlogConnector {
         };
 
         connector.register_as_replica().await?;
-        connector.request_binlog().await?;
-
+        let binlog_request = connector.request_binlog().await;
+        match binlog_request {
+            Ok(()) => (),
+            Err(mysql_async::Error::Server(ref err))
+                if mysql_srv::ErrorKind::from(err.code)
+                    == mysql_srv::ErrorKind::ER_MASTER_FATAL_ERROR_READING_BINLOG =>
+            {
+                // Requested binlog is not available, we need to re-snapshot
+                error!(error = %err, "Failed to request binlog");
+                return Err(ReadySetError::FullResnapshotNeeded);
+            }
+            Err(mysql_async::Error::Other(ref err)) => {
+                error!(error = %err, "Failed to request binlog");
+                if err.downcast_ref::<ReadySetError>() == Some(&ReadySetError::FullResnapshotNeeded)
```

Comment:
Should this conditional have an "else" branch? Right now, if the error isn't `ReadySetError::FullResnapshotNeeded`, we'll just continue with the snapshotting/replication process in spite of the request binlog error. It seems like we should propagate any other error upwards, which would prompt the replicator to restart

Reply:
@ethan-readyset good catch. I think we should enter this arm only if error is `mysql_async::Error::Other` && `ReadySetError::FullResnapshotNeeded` otherwise we fall back to default `Err` arm